# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/FuzzySecurityInfo.cpp
# Commit: 11e5baee978e
# Full Hash: 11e5baee978e8f98fa3795e430dd60c7600c9e8c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1580315
# File Overlap Count: 2
# Description:
#   Bug 1580315 - Convert certList to raw array for nsITransportSecurityInfo r=keeler,Ehsan,kershaw
#   
#   This patch converts the certList attribute of nsITransportSecurityInfo
#   from nsIX509CertList to Array<nsIx509Cert>
#   
# ==============================================================================

diff -r 341d72ef4bf5 -r 11e5baee978e netwerk/base/FuzzySecurityInfo.cpp
--- a/netwerk/base/FuzzySecurityInfo.cpp	Mon Oct 21 19:33:02 2019 +0000
+++ b/netwerk/base/FuzzySecurityInfo.cpp	Mon Oct 21 19:49:01 2019 +0000
@@ -38,7 +38,8 @@
 }
 
 NS_IMETHODIMP
-FuzzySecurityInfo::GetFailedCertChain(nsIX509CertList** _result) {
+FuzzySecurityInfo::GetFailedCertChain(
+    nsTArray<RefPtr<nsIX509Cert>>& aFailedCertChain) {
   MOZ_CRASH("Unused");
   return NS_OK;
 }
@@ -54,8 +55,8 @@
 }
 
 NS_IMETHODIMP
-FuzzySecurityInfo::GetSucceededCertChain(nsIX509CertList** _result) {
-  NS_ENSURE_ARG_POINTER(_result);
+FuzzySecurityInfo::GetSucceededCertChain(
+    nsTArray<RefPtr<nsIX509Cert>>& aSucceededCertChain) {
   MOZ_CRASH("Unused");
   return NS_OK;
 }